_posts/2016-12-06-the-state-of-babel.md

```diff
@@ -0,0 +1,460 @@
+
+Relevant Issues:
+
+- [Previous Linting APIs Discussion](https://github.com/babel/babel/issues/1829)
+- [Running smoke tests on ESLint](https://github.com/babel/babel-eslint/issues/62)
+- [babel/babel-eslint#88](https://github.com/babel/babel-eslint/issues/88) is still relevant now
+
+### Minification
+
+[Babili](https://github.com/babel/babili) is our new Babel-powered minifier, enabling you to produce minified code while targeting the latest browsers.
+
+**In**
+
+```js
+class Mangler {
+  constructor(program) {
+    this.program = program;
+  }
+}
+new Mangler(); 
+```
+
+**Out**
+
+```js
+// ES2015 code -> Babili -> Minified ES2015 Code
+class a{constructor(b){this.program=b}}new a;
+```
+
+Check out our [blog post](http://babeljs.io/blog/2016/08/30/babili) for more info.
+
+Since it was recently released, we're looking for some new contributors! There's a lot of small bugs and things that could be improved for those looking for a new project to help out with!
+
+### Codemods / Refactoring / `eslint --fix`
+
+A `codemod` is a tool assisted code modification; a fancy find-and-replace.
+
+If you wanted to change `merge({})` to `Object.assign({})` (and maybe later object rest) you might do a regex replace. But you don't want to replace the other parts of the code that also might be called `merge` such as imports/exports, strings, comments, and local variables. To do it safely you'll want something more powerful that only changes the specific code you need.
+
+Although Babel already handles transforming code into other code, being a compiler it doesn't really about the styling of the input code. By using Babel with it's default options to do a codemod you'll find that the `git diff` will look really messy.
+
+Enter [Recast](https://github.com/benjamn/recast): a tool that preserves the formatting of unmodified code while also pretty-printing any new AST nodes.
+
+![recast](https://cloud.githubusercontent.com/assets/588473/16584612/04a82078-428b-11e6-9f79-a665eef848ea.gif)
+
+> In the screenshot above, the top-left pane is the input code and the bottom-right pane is the output code after the babel plugin is run. In this case it's preserving the whitespace of the input code when possible.
+
+By passing in [Recast](https://github.com/benjamn/recast) in the options, Babel can power the future of codemods.
+
+`.babelrc`
+
+```js
+{
+  "parserOpts": {
+    "parser": "recast"
+  },
+  "generatorOpts": {
+    "generator": "recast"
+  }
+}
+```
+
+Run the relevant Babel transform(s) on the source code and overwrite it:
+
+```sh
+babel src -d src 
+```
+
+This feature was just made possible so we're looking forward to making it easier to use and seeing the transformations it can enable. Check out the [6.16.0 blog post](https://babeljs.io/blog/2016/09/28/6.16.0#new-feature) for more info!
+
+Other relevant projects: [JSCodeshift](https://github.com/facebook/jscodeshift), [Recast](https://github.com/benjamn/recast), [Lebab](https://github.com/lebab/lebab).
```

Comment:
Recast was already mentioned, so either remove “Other” from the start of the sentence or remove the duplicate link